src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java

```diff
@@ -472,6 +303,41 @@ public void deleteEntities(PersistenceEntity... entities) {
 	}
 
 	/**
+	 * This method executes a query in a certain transaction context while a session is open and returns the result or a default value.
+	 * @param session - the database session, which must be open
+	 * @param query - the query to execute and to retrieve the results from
+	 * @param transaction - the current transaction
+	 * @param getValue - the function to get the results from the query
+	 * @param defaultValue - a fall back default value in case anything went wrong or no entities were found
+	 * @param <R> - the result type
+	 * @param <Q> - the query type
+	 * @return - an object of the result value type
+	 */
+	protected <R, Q> R getEntities(Session session,
+								   Query<Q> query,
+								   Transaction transaction,
+								   Callable<R> getValue,
+								   R defaultValue) {
+
+		try {
+			R result = getValue.call();
+			transaction.commit();
+			return result;
+		} catch (NoResultException e) {
```

Comment:
and the breaks in the signature look strange, maybe there is a better way to break them